javagen/src/main/java/com/azure/autorest/util/MethodUtil.java

```diff
@@ -55,4 +70,155 @@ public static boolean isMethodIncludeRepeatableRequestHeaders(ProxyMethod proxyM
+        contentType.setDescription("The content type");
+        for (int i = 0; i < request.getParameters().size(); ++i) {
+            if (request.getParameters().get(i).getSchema() instanceof BinarySchema) {
+                contentType.setRequired(request.getParameters().get(i).isRequired());
+                break;
+            }
+        }
+        contentType.setImplementation(Parameter.ImplementationLocation.METHOD);
+        SealedChoiceSchema sealedChoiceSchema = new SealedChoiceSchema();
+        StringSchema stringSchema = new StringSchema();
+        stringSchema.setType(Schema.AllSchemaTypes.STRING);
+        sealedChoiceSchema.setChoiceType(stringSchema);
+        sealedChoiceSchema.setChoices(getContentTypeChoiceValues(requests));
+        sealedChoiceSchema.setType(Schema.AllSchemaTypes.fromValue("sealed-choice"));
+        Language language = new Language();
+        language.setName("contentType");
+        language.setSerializedName("Content-Type");
+        language.setDescription("The content type");
+        sealedChoiceSchema.setLanguage(new Languages());
+        sealedChoiceSchema.getLanguage().setJava(language);
+        sealedChoiceSchema.getLanguage().setDefault(language);
+        sealedChoiceSchema.getLanguage().setJava(language);
+        sealedChoiceSchema.setProtocol(new Protocols());
+        contentType.setSchema(sealedChoiceSchema);
+        contentType.setImplementation(Parameter.ImplementationLocation.METHOD);
+        contentType.setProtocol(new Protocols());
+        contentType.getProtocol().setHttp(new Protocol());
+        contentType.getProtocol().getHttp().setIn(RequestParameterLocation.HEADER);
+        contentType.setLanguage(new Languages());
+        contentType.getLanguage().setDefault(language);
+        contentType.getLanguage().setJava(language);
+        return contentType;
+    }
+
+    /**
+     *
+     * @param parameters a list of parameters
+     * @return return the index of the BinarySchema parameter, if not found, return -1
+     */
+    private static int getBinarySchemaBodyOrContentLengthParameterIndex(List<Parameter> parameters) {
+        int res = -1;
+        for(int i = 0; i < parameters.size(); ++i) {
+            if (parameters.get(i).getProtocol() != null && parameters.get(i).getProtocol().getHttp() != null
+                    && RequestParameterLocation.HEADER.equals(parameters.get(i).getProtocol().getHttp().getIn())
+                    && parameters.get(i).getLanguage() != null && parameters.get(i).getLanguage().getJava() != null
+                    && "Content-Length".equalsIgnoreCase(parameters.get(i).getLanguage().getJava().getSerializedName())) {
+                res = i;
+                break;
+            }
+        }
+        if (res == -1) {
+            for (int i = 0; i < parameters.size(); i++) {
+                if (parameters.get(i).getSchema() instanceof BinarySchema) {
+                    res = i;
+                }
+            }
```

Comment:
Why do we need to iterate over the list twice? Can we check for the schema type in the first iteration itself and return the appropriate index?